lib/limiter/reservation.go

```diff
@@ -0,0 +1,133 @@
+package limiter
+
+import (
+	"context"
+	"sync"
+	"tcplb/lib/core"
+)
+
+type ClientReservation struct {
```

Comment:
great comment, thank you.  maybe "reservation" isnt the cleanest abstraction to use. 

with a struct there is a place to accommodate adding stuff in future. but YAGNI for this scope of work.

i have removed this to simplify the API. it may be fractionally less odd now.

maybe taking the reservation concept too far, a client might have preferences about which upstream to connect to (one outside with view overlooking the river?). 

> it doesn't matter which reservation instance gets passed

maybe to support tcplb's commercial model, we could make reservations non-fungible :upside_down_face: